app/code/community/Bolt/Boltpay/controllers/ShippingController.php

```diff
@@ -343,35 +282,28 @@ public function getPrefetchCacheIdentifier($quote, $addressData)
             $cacheIdentifier .= '_' . $addressData['postcode'];
         }
 
+        if (isset($addressData['city'])) {
+            $cacheIdentifier .= '_' . $addressData['city'];
```

Comment:
I think you should always append these. otherwise 1) empty city and region 'foo' and 2) city 'foo' and empty region would have a same key

Reply:
yep.  perhaps an identifying label will help.